### PR TITLE
TEXT-39 WordUtils should use toXxxxCase(int) rather than toXxxxCase(char)

### PR DESCRIPTION
**Pull request by ameyjadiye**

### CODE REVIEW

**Comment by ameyjadiye** [comment]
@chtompki @PascalSchumacher @kinow , please review changes and let me know if more improvements needed here else I have verified everything on my end, all existing test cases are passing with changes.

**Comment by kinow** [comment]
Looks good I think. The code points is handled with the Character#charCount, so it increments with 1 or 2 (when it finds a surrogate pair). I noticed that if I pass a String such as `new String(bytes, "UTF-16")` the method returns a new `String(bytes)`, ignoring my charset... though I suspect it is smart enough to keep the current charset of the other chars? +1 for merging

File: src/main/java/org/apache/commons/text/WordUtils.java [hunk 1]
```diff
@@ -414,18 +414,29 @@ public static String capitalize(final String str, final char... delimiters) {
...
-        final char[] buffer = str.toCharArray();
+        int strLen = str.length();
+        int [] newCodePoints = new int[strLen];
+        int outOffset = 0;
+
         boolean capitalizeNext = true;
-        for (int i = 0; i < buffer.length; i++) {
-            final char ch = buffer[i];
-            if (isDelimiter(ch, delimiters)) {
+        for (int index = 0; index < str.length();) {
```
Comment (chtompki) [inline src/main/java/org/apache/commons/text/WordUtils.java]: `str.length();` could be replaced with `strLen` initialized above (just from a performance standpoint). Note, this is extra nit-picky and need not block the PR.

File: src/main/java/org/apache/commons/text/WordUtils.java [hunk 2]
```diff
@@ -537,18 +548,29 @@ public static String uncapitalize(final String str, final char... delimiters) {
...
-        final char[] buffer = str.toCharArray();
+        int strLen = str.length();
+        int [] newCodePoints = new int[strLen];
+        int outOffset = 0;
+
         boolean uncapitalizeNext = true;
-        for (int i = 0; i < buffer.length; i++) {
-            final char ch = buffer[i];
-            if (isDelimiter(ch, delimiters)) {
+        for (int index = 0; index < str.length();) {
```
Comment (chtompki) [inline src/main/java/org/apache/commons/text/WordUtils.java]: Same here with `str.length();`.

File: src/main/java/org/apache/commons/text/WordUtils.java [hunk 3]
```diff
@@ -578,30 +600,34 @@ public static String swapCase(final String str) {
...
+        final int strLen = str.length();
+        int [] newCodePoints = new int[strLen];
+        int outOffset = 0;
         boolean whitespace = true;
-
-        for (int i = 0; i < buffer.length; i++) {
-            final char ch = buffer[i];
-            if (Character.isUpperCase(ch)) {
-                buffer[i] = Character.toLowerCase(ch);
+        for (int index = 0; index < strLen;) {
```
Comment (chtompki) [inline src/main/java/org/apache/commons/text/WordUtils.java]: Right, this is what I was thinking.

File: src/main/java/org/apache/commons/text/WordUtils.java [hunk 4]
```diff
@@ -739,6 +765,27 @@ private static boolean isDelimiter(final char ch, final char[] delimiters) {
...
 
+  //-----------------------------------------------------------------------
+    /**
+     * Is the codePoint a delimiter.
+     *
+     * @param codePoint the codePint to check
+     * @param delimiters  the delimiters
+     * @return true if it is a delimiter
+     */
+    private static boolean isDelimiter(final int codePoint, final char[] delimiters) {
```
Comment (chtompki) [inline src/main/java/org/apache/commons/text/WordUtils.java]: Any reason to not make the signature here:
```java
public static String isDelimiter(final int codePoint, final char... delimiters);
```
I suppose we could have a debate about making this public v. private, but I'm not sure I see a reason for it to be exclusively private other than not needing to write as much documentation. I'm open to either way here, it was just something that crossed my mind.

**Comment by chtompki** [comment]
Am willing to merge. Just waiting to see what other folks think about what I said here.

File: src/main/java/org/apache/commons/text/WordUtils.java (hunk 4 above)
Comment (ameyjadiye) [inline src/main/java/org/apache/commons/text/WordUtils.java]: Thanks for pointing this, I created method first then generated it with eclipse by default it generated private so missed this one, I think this should be *public*.

**Comment by kinow** [comment]
>I'm open to either way here

Same here regarding make that method public  :-)

**Comment by ameyjadiye** [comment]
I think most of us are ok with making it ```public```, +1 to merge

**Comment by chtompki** [comment]
Looks good. Will pull this in tonight.

**Pull request merged**